schematic/utils/google_api_utils.py

```diff
@@ -2,14 +2,23 @@
 
 # pylint: disable=logging-fstring-interpolation
 
-import os
-import logging
 import json
-from typing import Any, Union, no_type_check, TypedDict
+import logging
+import os
+from typing import Any, Callable, TypedDict, Union, no_type_check
 
 import pandas as pd
-from googleapiclient.discovery import build, Resource  # type: ignore
 from google.oauth2 import service_account  # type: ignore
+from googleapiclient.discovery import Resource, build  # type: ignore
+from googleapiclient.errors import HttpError  # type: ignore
+from tenacity import (
```

Comment:
Oh cool, was tenacity already part of the dependencies?

Reply:
Yup! I took a portion of this decorator from another part of the schematic code base